### PR TITLE
[v2 rc] Async and Init cleanup

### PR DESCRIPTION
**Pull request by xzilja**
- Remove unnecessary async / awaits
- Add missing awaits
- Re-do initialisation (init) methods to not rely on events, add additional checks and throw errors where necessary

### CODE REVIEW

File: packages/utils/src/error.ts [hunk 1]
```diff
@@ -57,6 +57,8 @@ export const ERROR_TYPE = enumify({
   UNSUPPORTED_NOTIFICATION: "UNSUPPORTED_NOTIFICATION",
   UNSUPPORTED_SIGNAL: "UNSUPPORTED_SIGNAL",
   USER_DISCONNECTED: "USER_DISCONNECTED",
+  // 8000 (Not Initialized)
+  NOT_INITIALIZED: "NOT_INITIALIZED",
```
Comment (bkrem) [inline packages/utils/src/error.ts]: Doesn't this belong in the internal section i.e. `1xxx`? Why are we creating a new category for this?

File: packages/utils/src/error.ts [hunk 2]
```diff
@@ -549,6 +551,16 @@ export const ERROR: Record<ErrorType, Error> = {
       message: ERROR[ERROR_TYPE.USER_DISCONNECTED].stringify(params),
     }),
   },
+  // 8000 (Not Initialized)
+  [ERROR_TYPE.NOT_INITIALIZED]: {
+    type: ERROR_TYPE.NOT_INITIALIZED,
+    code: 8000,
+    stringify: (params?: any) => `Not Initialized ${params ? `: ${params?.toString()}` : ""}`,
```
Comment (bkrem) [inline packages/utils/src/error.ts]:
```suggestion
    stringify: (params?: any) => `${params ? `: ${params?.toString()}` : ""} was not initialized`,
```

File: packages/core/src/controllers/subscriber.ts [hunk 3]
```diff
@@ -144,29 +135,22 @@ export class Subscriber extends ISubscriber {
 
   private onEnable() {
     this.cached = [];
-    this.events.emit(SUBSCRIBER_EVENTS.enabled);
```
Comment (bkrem) [inline packages/core/src/controllers/subscriber.ts:147]: Are we sure this isn't needed? Are other controllers relying on this being emitted?

cc @pedrouid

Comment (xzilja) [reply packages/core/src/controllers/subscriber.ts:147]: Not needed, this was internal event signifying that initialization completed. All of these were replaced with a boolean flag instead and made consistent to not rely on events like this in all controllers. I'll also remove this one from the list of `SUBSCRIBER_EVENTS`

File: packages/utils/src/error.ts (hunk 1 above)
Comment (xzilja) [reply packages/utils/src/error.ts]: Moved to internal and changed code 👍

File: packages/utils/src/error.ts (hunk 2 above)
Comment (xzilja) [reply packages/utils/src/error.ts]: Updated 👍

**Pull request merged**